src/TypeScript/NodeJSLocator/NodeVersionInfoProvider.cs

```diff
@@ -19,48 +19,42 @@
  */
 
 using System;
+using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.Diagnostics;
 using System.IO.Abstractions;
-using SonarLint.VisualStudio.Integration;
 
 namespace SonarLint.VisualStudio.TypeScript.NodeJSLocator
 {
-    internal interface INodeLocator
+    internal interface INodeVersionInfoProvider
     {
-        /// <summary>
-        /// Returns the absolute file path of a compatible `node.exe`, or null if no compatible version was found.
-        /// </summary>
-        string Locate();
+        IEnumerable<NodeVersionInfo> GetAllNodeVersions();
     }
 
-    [Export(typeof(INodeLocator))]
+    [Export(typeof(INodeVersionInfoProvider))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class NodeLocator : INodeLocator
+    internal class NodeVersionInfoProvider : INodeVersionInfoProvider
     {
         private readonly INodeLocationsProvider nodeLocationsProvider;
-        private readonly ILogger logger;
         private readonly IFileSystem fileSystem;
         private readonly Func<string, Version> getNodeExeVersion;
 
         [ImportingConstructor]
-        public NodeLocator(INodeLocationsProvider nodeLocationsProvider, ILogger logger)
-            : this(nodeLocationsProvider, logger, new FileSystem(), GetNodeVersion)
+        public NodeVersionInfoProvider(INodeLocationsProvider nodeLocationsProvider)
+            : this(nodeLocationsProvider, new FileSystem(), GetNodeVersion)
         {
         }
 
-        internal NodeLocator(INodeLocationsProvider nodeLocationsProvider,
-            ILogger logger,
+        internal NodeVersionInfoProvider(INodeLocationsProvider nodeLocationsProvider,
             IFileSystem fileSystem,
             Func<string, Version> getNodeExeVersion)
         {
             this.nodeLocationsProvider = nodeLocationsProvider;
-            this.logger = logger;
             this.fileSystem = fileSystem;
             this.getNodeExeVersion = getNodeExeVersion;
         }
 
-        public string Locate()
+        public IEnumerable<NodeVersionInfo> GetAllNodeVersions()
```

Comment:
This class is now responsible for lazily locating all the node installations and calculating their versions.
Telemetry will call this.